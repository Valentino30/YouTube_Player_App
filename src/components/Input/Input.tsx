import { ChangeEvent } from "react";

type InputProps = {
  type?: string;
  value: string;
  placeholder?: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
};

export default function Input({ value, onChange, placeholder = "Type here", type = "text" }: InputProps) {
  return (
    <input
      type={type}
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      className="p-2 rounded-lg shadow-sm focus:outline-none focus:ring focus:ring-gray-700 mb-4 bg-gray-800"
    />
  );
}
