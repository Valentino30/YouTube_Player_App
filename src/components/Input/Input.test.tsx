import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import Input from "./Input";

describe("Input Component", () => {
  it("renders without errors", () => {
    render(<Input value="test" onChange={() => {}} />);
    expect(screen.getByDisplayValue("test")).toBeInTheDocument();
  });
  it("calls the onChange function when the value changes", async () => {
    const onChange = vi.fn();
    render(<Input value="test" onChange={onChange} />);
    const input = screen.getByDisplayValue("test");
    await userEvent.type(input, "new value");
    expect(onChange).toHaveBeenCalled();
  });
  it("has a placeholder when passed the placeholder prop", () => {
    render(<Input value="test" onChange={() => {}} placeholder="Type here" />);
    expect(screen.getByPlaceholderText("Type here")).toBeInTheDocument();
  });
  it("has a type of text when no type prop is passed", () => {
    render(<Input value="test" onChange={() => {}} />);
    expect(screen.getByDisplayValue("test")).toHaveAttribute("type", "text");
  });
  it("matches the snapshot", () => {
    const { asFragment } = render(<Input value="test" onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
