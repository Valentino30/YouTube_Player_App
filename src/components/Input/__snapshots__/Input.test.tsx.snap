// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Component > matches the snapshot 1`] = `
<DocumentFragment>
  <input
    class="p-2 rounded-lg shadow-sm focus:outline-none focus:ring focus:ring-gray-700 mb-4 bg-gray-800"
    placeholder="Type here"
    type="text"
    value="test"
  />
</DocumentFragment>
`;
